Added windows setup instructions

ajCameron opened the pull request:
Detailed instructions as to how to set up mewbot on a clean windows install.

- Suggested settings if installing python
- setting up a venv
- adding shortcuts to make venv activation easier
- Test running the linters and the tests
- adding mewbot to the PYTHONPATH so the user can run the examples
- running the examples

Setup, after following instructions, is tested and working.

ajCameron committed:
Added windows setup instructions

Detailed instructions as to how to set up mewbot on a clean windows install.

- Notes on the interpreter to use
- Suggested settings if installing python
- setting up a venv
- adding shortcuts to make venv activation easier
- Test running the linters and the tests
- adding mewbot to the PYTHONPATH so the user can run the examples
- running the examples

Setup, after following instructions, is tested and working.

exachixkitsune commented:
There is an alternative to adding to `PYTHONPATH`, which is to add:
`.\mewbot_venv\Lib\site-packages\mewbot.pth`
with a single line in it, being the path to your src directory.
(Say, you have mewbot in `C:\mewbot_dev`, you would have `mewbot.pth` being `C:\mewbot_dev\src`)

May be too complicate to include, but may be a neat alternative.

javajawa commented:
> There is an alternative to adding to `PYTHONPATH`, which is to add:
> 
> `.\mewbot_venv\Lib\site-packages\mewbot.pth`
> 
> with a single line in it, being the path to your src directory.
> 
> (Say, you have mewbot in `C:\mewbot_dev`, you would have `mewbot.pth` being `C:\mewbot_dev\src`)
> 
> 
> 
> May be too complicate to include, but may be a neat alternative.

I didn't know that, and I wonder if it is possible for the other platforms/OSes -- might make some things easier overall

ajCameron commented:
I definitely think it's worth including as an alternative - much more elegant that hacking on the activate script directly.

javajawa commented:
If it accepts relative paths (which it might, depending on the underlying implementation), we can even commit it to the codebase...

ajCameron committed:
Added .pth as the preferred option for getting tools on the path

- Modifying the activation script is still included for completeness, but is now advised against.

ajCameron commented:
Have included use of .pth to modify the interpreter path as the preferred method.

exachixkitsune commented on dev-docs/windows-dev-setup.md:
````diff
@@ -0,0 +1,462 @@
+C:\mewbot_dev\mewbot_venv\Lib\site-packages
+```
+
+Simple add a file named `mewbot.pth` here with the single line `C:\mewbot_dev\mewbot`. It's probably best to add it as a new file, instead of modifying one of the existing ones - as these are under system control and may be overwritten without warning.
````
```suggestion
Simple add a file named `mewbot.pth` here with the single line `C:\mewbot_dev\src`. It's probably best to add it as a new file, instead of modifying one of the existing ones - as these are under system control and may be overwritten without warning.
```
Pretty sure this should be as suggestion.

ajCameron replied:
Corrected

exachixkitsune commented on dev-docs/windows-dev-setup.md:
````diff
@@ -0,0 +1,462 @@
+Then open a new terminal, activate the venv, and check that `tools` can now be found by doing
+
+```shell
+(mewbot_venv) C:\Users\Adriann Ves Bloche>python
````
check; accidental name inclusion?

ajCameron replied:
Should be connected - thanks for the catch

exachixkitsune reviewed (changes requested):
Minor comments in file;
I will say, it's hard to read the induvidual sections - I often scrolled passed and missed them and the bold headings were more obvious.
Maybe make easy bit a heading, or an underline or something?

ajCameron committed:
Merge branch 'mainline' into dev_docs

ajCameron committed:
Update to docs to reflect current status of project

 - small change to linting script to deal with "python3" "python" name confusion
 - addressed accidental name inclusion (thank you @exachixkitsune !)
 - added details as to how to run linters and other utility scripts on windows (via git  sh/git bash)
 - tested new install instructions working

ajCameron committed:
Merge branch 'dev_docs' of https://github.com/mewler/mewbot into dev_docs

ajCameron committed:
Corrected error in .pth file spec

ajCameron commented:
Hi @exachixkitsune 

Made some changes to the linting/testing scripts so they should work on windows, as well as instructions as to how to run the .sh files on windows.

Hope this helps and resolves https://github.com/mewler/mewbot/issues/143

ajCameron committed:
Merge branch 'mainline' into dev_docs

ajCameron commented:
Superceded by https://github.com/mewbotorg/mewbot/pull/189
